linkerd/app/outbound/src/tcp/tests.rs

```diff
@@ -197,10 +199,20 @@ async fn resolutions_are_reused() {
 
     // Configure the mock destination resolver to just give us a single endpoint
     // for the target, which always exists and has no metadata.
-    let resolver = support::resolver().endpoint_exists(Addr::from(addr), addr, meta);
+    let resolver = support::resolver().endpoint_exists(
+        Addr::Name((svc_name.clone(), addr.port()).into()),
+        addr,
+        meta,
+    );
     let resolve_state = resolver.handle();
 
-    let profiles = support::profiles().profile(addr, Default::default());
+    let profiles = support::profiles().profile(
+        addr,
+        profile::Profile {
+            name: Some(svc_name),
+            ..profile::Profile::default()
+        },
+    );
```

Comment:
nit/tioli: the tests might be a _little_ more concise if we added a
```rust
pub fn named(name: &str) -> Profile {
    let name =  profile::Name::try_from(name.as_bytes()).unwrap();
    Profile {
         name: Some(name),
         ..Profile::default()
    }
}
```
or something in the test support `profiles` module? not a blocker though.

Reply:
...oh, i guess we also use the DNS name for addrs, so the let binding does make sense. _might_ be nice to have a test helper for `&str` -> `profile::Name` conversion, but not a blocker.